### PR TITLE
Pip fixes

### PR DESCRIPTION
**Pull request by manodeep**
Still fixing #208

### CODE REVIEW

**Comment by manodeep** [comment]
Huh - weird! I would have thought that the #209 would have already accounted for the initial commits on this branch.

**Comment by lgarrison** [comment]
Weird about the commits, maybe because we're "reusing" the same branch as the first PR?

**Comment by manodeep** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 2
           

Complexity decreasing per file
==============================
+ setup.py  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/manodeep/Corrfunc/pullRequest?prid=4914616&bid=16267130)

File: setup.py [hunk 1]
```diff
@@ -177,6 +177,15 @@ def replace_first_key_in_makefile(buf, key, replacement, outfile=None):
 def requirements_check():
     common_mk_file = pjoin(dirname(abspath(__file__)),
                            "common.mk")
+    # Since arbitrary python can be used even within the Makefile
+    # make sure that the current python executable is the same as the
+    # one specified in common.mk. Easiest way is to replace
+    this_python = sys.executable
+    key = "PYTHON"
+    replacement = '\n{0}:={1}'.format(key, this_python)
+    common = replace_first_key_in_makefile(common, key, replacement,
```
Comment (manodeep) [inline setup.py]: Codacy Issue found: [Using variable 'common' before assignment](https://app.codacy.com/manual/manodeep/Corrfunc/pullRequest?prid=4914616)

Comment (manodeep) [inline setup.py]: Codacy Issue found: [undefined name 'common'](https://app.codacy.com/manual/manodeep/Corrfunc/pullRequest?prid=4914616)

**Comment by manodeep** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity decreasing per file
==============================
+ setup.py  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/manodeep/Corrfunc/pullRequest?prid=4914616&bid=16267130)

**Comment by manodeep** [comment]
@lgarrison Thanks for fixing up the PR. Since the `common.mk` file has to be read in first any replacements, I moved the `this_python` substitution back down to after the python validation (i.e., where that section was previously)

**Comment by manodeep** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity decreasing per file
==============================
+ setup.py  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/manodeep/Corrfunc/pullRequest?prid=4914616&bid=16267130)

**Comment by manodeep** [comment]
@lgarrison Is this okay to merge?

(I am thinking of rebasing and merging - rather than the usual squash-merge)

**Comment by lgarrison** [comment]
Yes, looks good!  Either squash or rebase is fine with me, I don't understand the difference well enough to have a strong opinion!

**Pull request merged**